src/cascadia/UnitTests_TerminalCore/ConptyRoundtripTests.cpp

```diff
@@ -2975,3 +2980,368 @@ void ConptyRoundtripTests::NewLinesAtBottomWithBackground()
+
+            const auto actualNonSpacesAttrs = defaultAttrs;
+            const auto actualSpacesAttrs = rowCircled || isTerminal ? defaultAttrs : conhostDefaultAttrs;
+
+            VERIFY_ARE_EQUAL(isWrapped, tb.GetRowByOffset(row).GetCharRow().WasWrapForced());
+            if (isWrapped)
+            {
+                TestUtils::VerifyExpectedString(tb, std::wstring(charsInFirstLine, L'~'), til::point{ 0, row });
+            }
+            else
+            {
+                auto iter = TestUtils::VerifyExpectedString(tb, std::wstring(charsInSecondLine, L'~'), til::point{ 0, row });
+                TestUtils::VerifyExpectedString(std::wstring(width - charsInSecondLine, L' '), iter);
+            }
+        }
+    };
+
+    Log::Comment(L"========== Checking the host buffer state ==========");
+    verifyBuffer(*hostTb, si.GetViewport().ToInclusive());
+
+    Log::Comment(L"Painting the frame");
+    VERIFY_SUCCEEDED(renderer.PaintFrame());
+
+    Log::Comment(L"========== Checking the terminal buffer state ==========");
+    VERIFY_ARE_EQUAL(circledRows, term->_mutableViewport.Top());
+    verifyBuffer(*termTb, term->_mutableViewport.ToInclusive());
+}
+
+void doWriteCharsLegacy(SCREEN_INFORMATION& screenInfo, const std::wstring_view string, DWORD flags = 0)
+{
+    size_t dwNumBytes = string.size() * sizeof(wchar_t);
+    VERIFY_SUCCESS_NTSTATUS(WriteCharsLegacy(screenInfo,
+                                             string.data(),
+                                             string.data(),
+                                             string.data(),
+                                             &dwNumBytes,
+                                             nullptr,
+                                             screenInfo.GetTextBuffer().GetCursor().GetPosition().X,
+                                             flags,
+                                             nullptr));
+}
+
+void ConptyRoundtripTests::WrapNewLineAtBottomLikeMSYS()
+{
+    // See https://github.com/microsoft/terminal/issues/5691
+    Log::Comment(L"This test attempts to print text like the MSYS `less` pager "
+                 L"does. When it prints a wrapped line, we should make sure to "
+                 L"not break the line wrapping.");
+
+    // The importantly valuable variable here ended up being
+    // writingMethod=PrintWithWriteCharsLegacy. That was the one thing that
+    // actually repro'd this bug. The other variables were introduced as part of
+    // debugging, and are left for completeness.
+    BEGIN_TEST_METHOD_PROPERTIES()
+        TEST_METHOD_PROPERTY(L"Data:circledRows", L"{2, 4, 10}")
+        TEST_METHOD_PROPERTY(L"Data:paintEachNewline", L"{0, 1, 2}")
+        TEST_METHOD_PROPERTY(L"Data:writingMethod", L"{0, 1}")
+    END_TEST_METHOD_PROPERTIES();
+    constexpr int DontPaint = 0;
+    constexpr int PaintAfterBothLines = 1;
+    constexpr int PaintEveryLine = 2;

```

Comment:
I'm not totally clear what you mean by these cases here or above. I could reason it out with the logic below.... but again, more comments is probably more better.